### PR TITLE
PSA updates to simplify options for cloning packets

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
@hanw please take a look at this to see if it matches your expectations.

File: p4-16/psa/PSA.mdk [hunk 1]
```diff
@@ -295,11 +293,15 @@ the contents of several metadata fields in the struct
...
-        // TBD: Need a way to specify one (or more?) among multiple clone
-        // targets, if more than one such target exists.  Also to specify
-        // if anything is different about the cloned packet vs. other
-        // copies that might be made below.
+        create a copy of the packet and send it to the packet buffer,
+            after which it will start egress processing with
+            egress_port equal to the same value as the packet that was
+            cloned.
+        The contents of the packet will be the same as the packet as
+           it comes out of the egress deparser, prepended with any it
```
Comment (hanw) [inline p4-16/psa/PSA.mdk]: " prepended with any it", copy&paste error?

File: p4-16/psa/PSA.mdk [hunk 2]
```diff
@@ -295,11 +293,15 @@ the contents of several metadata fields in the struct
...
     }
     if (ostd.clone) {
-        create a copy of the packet to the clone target
-        // TBD: Need a way to specify one (or more?) among multiple clone
-        // targets, if more than one such target exists.  Also to specify
-        // if anything is different about the cloned packet vs. other
-        // copies that might be made below.
+        create a copy of the packet and send it to the packet buffer,
+            after which it will start egress processing with
+            egress_port equal to the same value as the packet that was
```
Comment (hanw) [inline p4-16/psa/PSA.mdk]: egress_port (clone_port?) of the cloned packet can be set to any port, and should not be restricted to the egress_port of the original packet coming into the egress pipeline.

File: p4-16/psa/PSA.mdk (hunk 1 above)
Comment (jafingerhut) [reply p4-16/psa/PSA.mdk]: Good catch.  Should be fixed.

File: p4-16/psa/PSA.mdk (hunk 2 above)
Comment (jafingerhut) [reply p4-16/psa/PSA.mdk]: ingress2egress clone allows cloning to CPU port.  Should egress2egress clone allow or prohibit this possibility?

Comment (hanw) [reply p4-16/psa/PSA.mdk]: It should allow this possibility.

Comment (jafingerhut) [reply p4-16/psa/PSA.mdk]: Done.

**Comment by hanw** [comment]
Can we merge this?

**Pull request merged**